src/net/sourceforge/kolmafia/persistence/ConcoctionDatabase.java

```diff
@@ -1255,7 +1256,15 @@ public static final synchronized void refreshConcoctionsNow() {
     Preferences.increment("_concoctionDatabaseRefreshes");
     ConcoctionDatabase.refreshNeeded = false;
 
-    List<AdventureResult> availableIngredients = ConcoctionDatabase.getAvailableIngredients();
+    List<AdventureResult> availableIngredientsList = ConcoctionDatabase.getAvailableIngredients();
+
+    // In addition to the list, we create a second data structure here for better performance.
+    // Because we do many lookups to the available ingredients to see how many there are,
+    // having an O(1) lookup helps a lot.
+    Map<Integer, AdventureResult> availableIngredients = new ArrayListMap<>();
```

Comment:
I think we could specify the initial size of this map -- it's the size of the populating list

Reply:
Yep, good idea.